engine/lib/phx/src/ui/hmgui/container.rs

```diff
@@ -0,0 +1,460 @@
+use std::cell::RefMut;
+use std::collections::HashMap;
+use std::fs::File;
+
+use glam::{Vec2, Vec4};
+
+use super::*;
+
+#[derive(Debug, Copy, Clone, Default, PartialEq)]
+pub enum LayoutType {
+    #[default]
+    None,
+    Stack,
+    Horizontal,
+    Vertical,
+}
+
+#[derive(Clone, Default, PartialEq)]
+pub struct HmGuiContainer {
+    pub children: Vec<Rf<HmGuiWidget>>,
+
+    // Layout
+    pub layout: LayoutType,
+    pub children_horizontal_alignment: AlignHorizontal,
+    pub children_vertical_alignment: AlignVertical,
+    pub padding_lower: Vec2,
+    pub padding_upper: Vec2,
+    pub spacing: f32,
+
+    pub children_hash: u32,
+    pub focus_style: FocusStyle,
+    pub offset: Vec2,
+    pub max_size: Vec2, // TODO: do we still need this?
```

Comment:
Can we answer this todo before merging, or is it something for later? If it's for later, lets track it somewhere as part of an issue.

Reply:
Not used at the moment - removed